source/common/router/rds_impl.h

```diff
@@ -129,8 +130,8 @@ class RdsRouteConfigProviderImpl
   RdsRouteConfigProviderImpl(
       const envoy::config::filter::network::http_connection_manager::v2::Rds& rds,
       const std::string& manager_identifier, Server::Configuration::FactoryContext& factory_context,
-      const std::string& stat_prefix,
-      RouteConfigProviderManagerImpl& route_config_provider_manager);
+      const std::string& stat_prefix, RouteConfigProviderManagerImpl& route_config_provider_manager,
+      SystemTimeSource& system_time_source);
```

Comment:
Could the `SystemTimeSource` be delivered via `factory_context` here?

Reply:
That's a cool idea. If we bundle the time_source references (maybe both System and Monotonic) we can simplify the signatures of
 - RouteConfigProviderManager::getRdsRouteConfigProvider()
 - RdsRouteConfigProviderImpl::RdsRouteConfigProviderImpl()
 - RouteConfigProviderUtil::create()
 - RouteConfigProviderManagerImpl::getRdsRouteConfigProvider()
 - RdsRouteConfigProviderImpl::RdsRouteConfigProviderImpl()
which would make this PR a little simpler.